include/deal.II/base/mpi_consensus_algorithms.templates.h

```diff
@@ -715,9 +824,23 @@ namespace Utilities
 
       template <typename T1, typename T2>
       std::vector<unsigned int>
-      Serial<T1, T2>::run()
+      Serial<T1, T2>::run(
+        const std::vector<unsigned int> &targets,
+        const std::function<std::vector<T1>(const unsigned int)>
+          &create_request,
+        const std::function<std::vector<T2>(const unsigned int,
+                                            const std::vector<T1> &)>
+          &answer_request,
+        const std::function<void(const unsigned int, const std::vector<T2> &)>
+          &             process_answer,
+        const MPI_Comm &comm)
       {
-        const auto targets = this->process.compute_targets();
+        (void)comm;
+        Assert((Utilities::MPI::job_supports_mpi() == false) ||
+                 (Utilities::MPI::n_mpi_processes(comm) == 1),
+               ExcMessage("You shouldn't use the 'Serial' class on "
+                          "communicators that have more than one process "
+                          "associated with it."));
```

Comment:
This is strictly speaking an unrelated addition, but I thought it was a good idea while I was already working on this function.